### ISSUE TITLE
reloading hugsql fails

### ISSUE THREAD

**Issue by opqdonut** [layerware/hugsql#65]
```
=> (require 'hugsql.core :reload)

CompilerException clojure.lang.ArityException: Wrong number of args (4) passed to: core/def-sqlvec-fns, compiling:(hugsql/core.clj:342:11) 
```

It seems to me like the arity 1 version of `def-sqlvec-fns` is trying to call a nonexistent arity 4 version [here]. Couldn't `&form &env` just be removed?

[here]: https://github.com/layerware/hugsql/blob/3e443afefd3550770527a17526aa607b3fbe7385/hugsql-core/src/hugsql/core.clj#L363

**Comment by csummers** [comment]
The `&form` and `&env` are necessary for multiple arity macros.

I think the real issue is that reloading namespaces in Clojure is very problematic.  Read through the tools.namespace comments found here: https://github.com/clojure/tools.namespace#reloading-code-motivation

**Comment by opqdonut** [comment]
Yes, I'm familiar with tools.namespace.

It seems reloading multi-arity macros is broken in general in clojure.

A workaround is to define the macro like

```clojure
(defmacro def-sqlvec-fns
  ([file] `(def-sqlvec-fns ~file {}))
  ([file options]
     ...)

**Comment by opqdonut** [comment]
See http://dev.clojure.org/jira/browse/CLJ-2136

**Comment by opqdonut** [comment]
See PR #66

**Comment by danielcompton** [comment]
I've run into this too, what is the recommended method for reloading functions defined by Hugsql? I'm using the reloaded pattern with tools.namespace for reloading code.

**Comment by dekz** [comment]
Any change we can get a release for the merged PR?

**Comment by csummers** [comment]
Closing...this was released in 0.4.8
